src/bin/extensions.ml

```diff
@@ -0,0 +1,125 @@
+(* This file is free software, part of dolmen. See file "LICENSE" for more information *)
+
+type t = string
+(** We only store the name of the extension. *)
+
+let builtin_extensions =
+  (* Note: this must be computed at toplevel in order to make sure we don't
+     pick up extensions loaded from plugins as builtin. *)
+  let builtin_extensions = Hashtbl.create 17 in
+  (* Add builtin typing extensions. *)
+  Dolmen_loop.Typer.Ext.iter (fun ext ->
+    Hashtbl.replace builtin_extensions (Dolmen_loop.Typer.Ext.name ext) ()
+  );
+  (* Add builtin model extensions (note: these may share name with typing
+     extensions). *)
+  Dolmen_model.Ext.iter (fun ext ->
+    Hashtbl.replace builtin_extensions (Dolmen_model.Ext.name ext) ()
+  );
+  builtin_extensions
+
+let pp ppf plugin =
+  let pp_location ppf plugin =
+    if Hashtbl.mem builtin_extensions plugin
+    then
+      Fmt.pf ppf "@ (builtin)"
+  in
+  Fmt.pf ppf "@[%s%a@]" plugin pp_location plugin
+
+let name plugin = plugin
+
+let list () =
+  let all_extensions =
+    Hashtbl.fold
+      (fun extension_name () exts ->
+        extension_name :: exts)
+      builtin_extensions
+      []
+  in
+  let all_extensions =
+    List.fold_left (fun all_extensions plugin ->
+      if Hashtbl.mem builtin_extensions plugin
```

Comment:
In that case, we could emit a warning to warn that an installed plugin with the same name as a builtin plugin will never be usable, as the builtin will always be loaded instead ? Even if it is only triggered when explicitly asking for the list of all available plugins with the command line, I think that would be useful.